pkg/internal/codegen/parse/index.go

```diff
@@ -183,31 +190,80 @@ type resourceTags struct {
 	ShortName string
 }
 
-// ParseResourceTag parses the tags in a "+resource=" comment into a resourceTags struct
-func parseResourceTag(tag string) resourceTags {
-	result := resourceTags{}
+//helperResourceAnnotation is a helper function to extract resource annotation.
+func helperResourceAnnotation(tag string, count int) (resourceTags, int, error) {
```

Comment:
this is not a good function name.  The `helper` doesn't convey anything.  The old name (`parseResourceAnnotation`) was much clearer.  Maybe `parseResourceAnnotationValue`?